ls1intum/Artemis
# `Development`: Unify handling text, modeling, and file upload submissions

### Checklist
#### General
#### Server

### Motivation and Context
This PR unifies how text and modeling submissions are handled. There were a couple small differences so far.
To review the changes, I suggest you checkout the branch and compare `ModelingSubmissionService` and `TextSubmissionService` next to each other in IntelliJ

After the unification, there is some code duplication now. In a future PR, we can try to extract the duplicated code, however due to the sub types of exercise and submission, this is an easy task and should therefore be done in a follow-up

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Course with a text, modeling, and file upload exercise
- 1 Exam with a text, modeling, and file upload exercise

1. Participate in the text, modeling and file upload exercises and submit changes multiple times within a course and within an exam2. 
2. Make sure the submissions are saved correctly, e.g. by reloading the page

### Review Progress





#### Code Review
#### Manual Tests
